packages/client-core/src/hooks/useEngineCanvas.ts

```diff
@@ -59,12 +55,56 @@ export const useEngineCanvas = (ref: React.RefObject<HTMLElement>) => {
     observer.observe(parent)
 
     return () => {
-      destroySpatialViewer()
       observer.disconnect()
       parent.removeChild(canvas)
-      originalParent?.appendChild(canvas)
     }
-  }, [lastRef.value, engineState.localFloorEntity, engineState.originEntity])
+  }, [lastRef.value])
+
+  /** Essentially mount/unmount upon the attach/detatch state of the ref node */
+  useEffect(() => {
+    if (!lastRef.value) return
+    const canvas = document.getElementById('engine-renderer-canvas') as HTMLCanvasElement
+    initializeSpatialViewer(canvas)
+    return () => {
+      destroySpatialViewer()
+    }
+  }, [!!lastRef.value])
+
+  /**
+   * Since the viewer and XR reference spaces can technically exist without the other,
+   * we need to reactively update the core renderer's scenes
+   */
+  const { viewerEntity, originEntity, localFloorEntity } = useMutableState(EngineState).value
+
+  useEffect(() => {
+    if (!viewerEntity || !originEntity) return
+
+    const rendererComponent = getOptionalMutableComponent(viewerEntity, RendererComponent)
+    if (!rendererComponent) return
+
+    rendererComponent.scenes.merge([originEntity])
+
+    return () => {
+      if (!hasComponent(viewerEntity, RendererComponent)) return
+      const index = rendererComponent.scenes.value.indexOf(originEntity)
+      rendererComponent.scenes[index].set(none)
+    }
+  }, [viewerEntity, originEntity])
+
+  useEffect(() => {
```

Comment:
I think this got duplicated somehow

Reply:
nope, one is local floor one is origin :)